### PR TITLE
apiVersion test migrated to devfileV2

### LINKED ISSUE #3794: API version and schema version tests should be migrated to devfileV2
/kind cleanup

https://github.com/openshift/odo/blob/master/pkg/devfile/parser/context/apiVersion_test.go
https://github.com/openshift/odo/blob/master/pkg/devfile/parser/context/schema_test.go

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**
/kind cleanup

**What does does this PR do / why we need it**:
Migrate api-version test to devfileV2
**Which issue(s) this PR fixes**:

Fixes #3794 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Review by mik-dass — APPROVED** [review]
/lgtm

File: pkg/devfile/parser/context/apiVersion_test.go [hunk 1]
```diff
@@ -9,8 +9,8 @@ import (
 func TestSetDevfileAPIVersion(t *testing.T) {
 
 	const (
-		apiVersion          = "1.0.0"
-		validJson           = `{"apiVersion": "1.0.0"}`
+		apiVersion          = "2.0.0"
+		validJson           = `{"apiVersion": "2.0.0"}`
```
Comment (adisky) [inline pkg/devfile/parser/context/apiVersion_test.go:13]: it is schema version for devfile v2, so valid json would be `{"schemaVersion": "2.0.0"}`

**Review by adisky — COMMENTED** [review]
it is schema version for devfile v2, so valid json would be {"schemaVersion": "2.0.0"}

**Comment by girishramnani** [comment]
are we sure these are all the tests that need migration?

**Comment by anandrkskd** [comment]
@girishramnani  according to the issue #3794 two files needed to be migrated to v2 and one of the file(schema_test.go) was fixed by PR #3867.

**Pull request merged**